fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACAuthorizingRealm.java

```diff
@@ -72,4 +166,25 @@ protected AuthenticationInfo doGetAuthenticationInfo(final AuthenticationToken t
     public boolean supports(final AuthenticationToken token) {
         return false;
     }
+
+    private FedoraResource getResourceOrParentFromPath(final String path) {
```

Comment:
@whikloj Yes, it is still responsible for gathering up the authorizations and constructing the map of agent => modes. For now I just kept the logic in it and undisturbed, but it would likely be good to eventually fold the necessary bits of it into the WebACAuthorizingRealm.

Reply:
Cool, it can be removed from one or the other. I was thinking specifically of https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java#L131

Reply:
@whikloj To remove that method, it would probably be best to make the WebACRolesProvider no longer implement the `AccesRolesProvider` and just be standalone class.

Reply:
@whikloj Regarding my last comment, scrubbing a lot of the now unnecessary WebACRolesProvider bits related to Modeshapes authz system should wait until the completion of https://jira.duraspace.org/browse/FCREPO-2763 and the removal of the FAD from the spring config.